### ISSUE TITLE
The required role for using the "endpoint" attribute to connect to Azure App Configuration is not documented.

### ISSUE THREAD

**Issue by jimmyca15** [aspnet/MicrosoftConfigurationBuilders#102]
the "-rc" release of the Azure App Configuration configuration builder introduced the endpoint attribute which is used to set up a connection to an App Configuration instance. This connection is made using AAD auth and requires the identity to be assigned to one of either the Azure App Configuration Data Reader or Azure App Configuration Data Owner roles. This requirement should be documented in the readme where the endpoint parameter is explained.

Specifically it is very hard to figure out
1. What identity is used
  * We need to know this to make the role assignment
2. What role assignment should be made for that identity.

**Comment by stilren** [comment]
This is how to make the role assignment using ARM templates:
`{
    "name": "[concat(parameters('configName'), '/Microsoft.Authorization/', guid(uniqueString(parameters('appServiceName'))))]",
    "type": "Microsoft.AppConfiguration/configurationStores/providers/roleAssignments",
    "apiVersion": "2018-09-01-preview",
    "properties": {
        "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', '516239f1-63e1-4d78-a4de-a74fb236a071')]",
        "principalId": "[reference(parameters('appServiceName'),'2016-08-01','Full').identity.principalId]"
    }
}`

**Comment by rafaeldca** [comment]
*Do you plan or already support* applications running in VM's ? How do we authenticate the endpoint when running the application in my dev machine ?

I'm working with a few **ASP.NET** applications and new applications are being developed in **ASP.NET core (.NET core)**, our team is investigating the use of **Azure App Configuration** but so far could not connect the service.

I am using the packages below:
 ```
<package id="Microsoft.Configuration.ConfigurationBuilders.AzureAppConfiguration" version="1.0.0-rc" targetFramework="net48" />
<package id="Microsoft.Configuration.ConfigurationBuilders.Base" version="2.0.0-rc" targetFramework="net48" />
<package id="Microsoft.Configuration.ConfigurationBuilders.Environment" version="2.0.0-rc" targetFramework="net48" />
```

**Comment by StephenMolloy** [comment]
@rafaeldca for Azure App Configuration, you should be able to use a connection string again for connecting to the service when the RTM package hits nuget.

If you use just `endpoint`, your identity is determined by Azure.Identity.DefaultAzureCredential. This is magically you in VS, and magically your service identity I think if running in Azure. I have pinged someone in the Azure group for a comment on how to setup the correct environment for DefaultAzureCredential to do the thing you want in #103.

As for the original doc issue, fixed with #108
